### ISSUE TITLE
false positive: use of moved value

### ISSUE THREAD

**Issue by niveau0** [intellij-rust/intellij-rust#3270]
## Environment

* **Intellij-Rust plugin version:** 0.2.0.2114-183
* **Rust toolchain version:** 1.31.1
* **IDE name and version:** IDEA Community 2018.3.3
* **Operating system:** Ubuntu 18.04

## Problem description
Displays false positive "Use of moved value", where cargo build works without warnings. See simplified demo code.

## Steps to reproduce

```rust
fn demo() -> Result<Vec<i32>, io::Error> {
    let vec = vec![1,2,4,5];

    let res : Result<Vec<i32>, io::Error> = Ok(vec);
    
    if let Ok(mut vec) = res {
        vec.sort();
        Ok(vec)
    } else {
        res //<-- false positive here
    }
}
```
